images/metering-ansible-operator/roles/meteringconfig/tasks/deploy_resource.yml

```diff
@@ -0,0 +1,46 @@
+---
+
+- name: Preparing to {{ create | ternary('create', 'delete') }} {{ apis | map(attribute='kind') | join(', ') }} resources from {{ template_file }}
+  debug:
+    msg: Preparing to {{ create | ternary('create', 'delete') }} {{ apis | map(attribute='kind') | join(', ') }} resources from {{ template_file }}
+
+- when: create
+  block:
+    - name: Helm template {{ template_file }}
+      shell: helm template {{ meteringconfig_chart_path }} --name {{ release_name }} --namespace {{ namespace}} -x {{ template_file }} -f {{ values_file }}
+      register: result
+      no_log: true
+      ignore_errors: true
+
+    - name: Fail if helm template errors on {{ template_file }}
+      fail: msg="{{ result }}"
+      when: result.rc != 0
+
+    - name: Create resources from {{ template_file }}
+      vars:
+        to_create_resources: "{{ result.stdout | from_yaml_all | list }}"
+      loop: "{{ to_create_resources }}"
+      when: item is not none
+      include_tasks: create_resource.yml
+
+- when: not create
+  block:
+    - name: Query for {{ apis | map(attribute='kind') | join(', ') }} resources with selector {{ label_selector}} to delete
+      vars:
```

Comment:
`when` conditionals are typically placed at the bottom of blocks. https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html